clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -78,6 +79,8 @@
             "Last committed offset.");
     private static final Field.Int64 LOG_START_OFFSET = new Field.Int64("log_start_offset",
             "Earliest available offset.");
+    private static final Field.Int32 PREFERRED_READ_REPLICA = new Field.Int32("preferred_read_replica",
```

Comment:
Do we consider upgrading `FetchResponse` to automated protocol first to reduce our effort here?

Reply:
Yes, I believe that's on a list of TODOs regarding the protocol generator. As I understand it, there are some complexities with FetchRequest/Response that would need to be figured out.

Reply:
Yeah, `FetchResponse` will most likely be the last one to convert because we'll have to figure out how zero-copy will work.